core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +76,9 @@ public void loadNext() throws IOException {
     numRecordsRemaining--;
     if (numRecordsRemaining == 0) {
       in.close();
-      file.delete();
+      if (!file.delete()) {
+        logger.error("Unable to delete spill file {}", file.getPath());
```

Comment:
Error to me means something unrecoverable; the process can't continue. This is a warning maybe. It's maybe better to look across the code at places where `File.delete()` may fail silently and do something similar. And then open a JIRA for this small cross cutting change.